File: Server/route/userGroupChallenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserGroupChallengeRoute(r *gin.Engine) {
	userChallenge := r.Group("/userGroupChallenge")
	userChallenge.Use(middleware.JWTAuthMiddleware())
	{
		userChallenge.POST("/GetUserGroupChallengeList", api.GetUserGroupChallengeList)
		userChallenge.POST("/ResetUserGroupChallenge", api.ResetUserGroupChallenge)
		userChallenge.POST("/CheckGroupChallengeFlag", api.CheckGroupChallengeFlag)
	}
}

File: Web/src/api/userGroupChallenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetUserGroupChallengeList(data) {
  return apiClient.post("/userGroupChallenge/GetUserGroupChallengeList", data).then(handleResponse).catch(handleError)
}

function ResetUserGroupChallenge(data) {
  return apiClient.post("/userGroupChallenge/ResetUserGroupChallenge", data).then(handleResponse).catch(handleError)
}

function CheckGroupChallengeFlag(data) {
  return apiClient.post("/userGroupChallenge/CheckGroupChallengeFlag", data).then(handleResponse).catch(handleError)
}

const userGroupChallengeApi = {
  GetUserGroupChallengeList,
  ResetUserGroupChallenge,
  CheckGroupChallengeFlag
}

export default userGroupChallengeApi
